lib/elixir/src/elixir_bitstring.erl

```diff
@@ -323,6 +325,8 @@ format_error({unaligned_bitstring_in_match, Expr}) ->
     "    <<\"foo\", <<field, rest::bitstring>>::bitstring>>\n\n"
     "Got: ~ts",
   io_lib:format(Message, ['Elixir.Macro':to_string(Expr)]);
+format_error(pin_operator_on_left_side) ->
+  "^ operator is allowed only on the right side of the <> operator";
```

Comment:
Maybe the message is not clear enough for cases like:
```
"a" <> ^a <> "b"
```

Reply:
@ggcampinho what confuse me as non-english speaker is that your statement is an informative way instead of telling me what I am doing wrong like.

`^ operator should be in the right side of the <> operator only`

Reply:
I'd think it would be interesting to say something on the lines of 

```elixir
"""
There should not be any variable-size binary before the pinned value as it's size cannot be verified.

The following example is invalid:
  a <> ^b = foo

The following example is valid
  "Known size string" <> ^a <> b = foo
"""
```

Naturally with better examples and some additional text explaining this limitation

Reply:
I tried to change a little bit and be more clear, can you check?

Your last example is invalid, and I think it's nice to show examples, but in this case, it won't help much to fix the issue, since swapping arguments will change the behavior. The documentation of the <> operator is very clear, so I think it's enough.